lib/entry.c

```diff
@@ -29,19 +29,25 @@ void pw_entry (const char *name, struct passwd *pwent)
 		pwent->pw_name = NULL;
```

Comment:
Isn't this a leak too?

Reply:
After refactoring sulogic.c and pw_entry(), I see that the caller immediately exits after that, so it's a short-lived leak, but still a leak.

I'm thinking that I'll resolve this by refactoring pw_entry() to return a -1 on error, so you can just ignore this.  Or you can refactor it yourself if you want.  :)

Reply:
Yes, it is a leak. And it is short-lived, but still a leak. I have adjusted my commit and added you as a co-author. Thank you for pointing it out!

Reply:
Thanks!